homeassistant/components/hassio/__init__.py

```diff
@@ -898,47 +908,75 @@ async def force_info_update_supervisor(self) -> None:
         self.hass.data[DATA_SUPERVISOR_INFO] = await self.hassio.get_supervisor_info()
         await self.async_refresh()
 
-    async def force_data_refresh(self) -> None:
+    async def force_data_refresh(self, first_update: bool) -> None:
         """Force update of the addon info."""
+        data = self.hass.data
+        hassio = self.hassio
         (
-            self.hass.data[DATA_INFO],
-            self.hass.data[DATA_CORE_INFO],
-            self.hass.data[DATA_CORE_STATS],
-            self.hass.data[DATA_SUPERVISOR_INFO],
-            self.hass.data[DATA_SUPERVISOR_STATS],
-            self.hass.data[DATA_OS_INFO],
+            data[DATA_INFO],
+            data[DATA_CORE_INFO],
+            data[DATA_CORE_STATS],
+            data[DATA_SUPERVISOR_INFO],
+            data[DATA_SUPERVISOR_STATS],
+            data[DATA_OS_INFO],
         ) = await asyncio.gather(
-            self.hassio.get_info(),
-            self.hassio.get_core_info(),
-            self.hassio.get_core_stats(),
-            self.hassio.get_supervisor_info(),
-            self.hassio.get_supervisor_stats(),
-            self.hassio.get_os_info(),
+            hassio.get_info(),
+            hassio.get_core_info(),
+            hassio.get_core_stats(),
+            hassio.get_supervisor_info(),
+            hassio.get_supervisor_stats(),
```

Comment:
In a future PR we can avoid the supervisor and core stats as well if we don't have any entities that want them

Reply:
https://github.com/home-assistant/core/pull/102362